pkg/compact/retention_test.go

```diff
@@ -244,17 +247,28 @@ func TestApplyRetentionPolicyByResolution(t *testing.T) {
 			metaFetcher, err := block.NewMetaFetcher(logger, 32, bkt, "", nil)
 			testutil.Ok(t, err)
 
-			if err := compact.ApplyRetentionPolicyByResolution(ctx, logger, bkt, metaFetcher, tt.retentionByResolution); (err != nil) != tt.wantErr {
+			blocksMarkedForDeletion := prometheus.NewCounter(prometheus.CounterOpts{})
+			if err := compact.ApplyRetentionPolicyByResolution(ctx, logger, bkt, metaFetcher, tt.retentionByResolution, blocksMarkedForDeletion); (err != nil) != tt.wantErr {
 				t.Errorf("ApplyRetentionPolicyByResolution() error = %v, wantErr %v", err, tt.wantErr)
 			}
 
 			got := []string{}
+			gotMarkedBlocksCount := 0.0
 			testutil.Ok(t, bkt.Iter(context.TODO(), "", func(name string) error {
-				got = append(got, name)
+				exists, err := bkt.Exists(ctx, filepath.Join(name, metadata.DeletionMarkFilename))
+				if err != nil {
+					return err
+				}
+				if !exists {
+					got = append(got, name)
+				} else {
```

Comment:
You don't need `else` just `return` in case above

Reply:
the else is required to increment the gotMarkedBlocksCount variable.

I am using this to check if the prometheus metric for blocks marked for deletion count is the same as what we expect

Reply:
it's not need you can just return in this case